Source/WebCore/workers/service/ServiceWorkerContainer.cpp

```diff
@@ -137,6 +137,7 @@ ServiceWorker* ServiceWorkerContainer::controller() const
 
 void ServiceWorkerContainer::addRegistration(const String& relativeScriptURL, const RegistrationOptions& options, Ref<DeferredPromise>&& promise)
 {
+    // FIXME: Below we seem to assume this exists. Should we assert context and change it to a reference?
```

Comment:
I would like reviewer feedback on this.

Reply:
Please do not add this comment. Pretty sure that if context was null then `m_isStopped` would be true so we'd return early on the next if check. Therefore, asserting that the context is non-null before the if check could cause crashes for no good reason.

Maybe we simply want to do:
```
if (m_isStopped || !context) {
    promise->reject(Exception(ExceptionCode::InvalidStateError));
    return;
}
```

For clarity.

Reply:
Sure, you can move it down too